test/saithrift/tests/saiacl.py

```diff
@@ -15,114 +15,427 @@
-        sai_thrift_create_route(self.client, vr_id, addr_family, ip_addr1, ip_mask1, nhop1)
-
-        # send the test packet(s)
-        pkt = simple_tcp_packet(eth_dst='00:77:66:55:44:33',
-                                eth_src='00:22:22:22:22:22',
-                                ip_dst='10.10.10.1',
-                                ip_src='192.168.0.1',
-                                ip_id=105,
-                                ip_ttl=64)
-        exp_pkt = simple_tcp_packet(
-                                eth_dst='00:11:22:33:44:55',
-                                eth_src='00:77:66:55:44:33',
-                                ip_dst='10.10.10.1',
-                                ip_src='192.168.0.1',
-                                ip_id=105,
-                                ip_ttl=63)
+        self.vr_id = sai_thrift_create_virtual_router(self.client, v4_enabled, v6_enabled)
+
+        self.rif_id1 = sai_thrift_create_router_interface(self.client, self.vr_id, 1, port1, 0, v4_enabled, v6_enabled, mac)
+        self.rif_id2 = sai_thrift_create_router_interface(self.client, self.vr_id, 1, port2, 0, v4_enabled, v6_enabled, mac)
+        self.rif_id3 = sai_thrift_create_router_interface(self.client, self.vr_id, 1, port3, 0, v4_enabled, v6_enabled, mac)
+
+        self.addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        self.ip_addr1 = IP_DST
+        self.ip_mask1 = IP_MASK
+        self.dmac1 = rewrite_mac2
+        sai_thrift_create_neighbor(self.client, self.addr_family, self.rif_id1, self.ip_addr1, self.dmac1)
+        self.nhop1 = sai_thrift_create_nhop(self.client, self.addr_family, self.ip_addr1, self.rif_id1)
+        sai_thrift_create_route(self.client, self.vr_id, self.addr_family, self.ip_addr1, self.ip_mask1, self.nhop1)
+        
+        wait_till_configuration_will_end()
+
+    def cleanupRouting(self):
+        sai_thrift_remove_route(self.client, self.vr_id, self.addr_family, self.ip_addr1, self.ip_mask1, self.nhop1)
+        self.client.sai_thrift_remove_next_hop(self.nhop1)
+        sai_thrift_remove_neighbor(self.client, self.addr_family, self.rif_id1, self.ip_addr1, self.dmac1)
+
+        self.client.sai_thrift_remove_router_interface(self.rif_id1)
+        self.client.sai_thrift_remove_router_interface(self.rif_id2)
+        self.client.sai_thrift_remove_router_interface(self.rif_id3)
+
+        self.client.sai_thrift_remove_virtual_router(self.vr_id)
+
+    def setupAclTable(self, stage = None, mac_src = None, mac_dst = None, mac_src_mask = None, mac_dst_mask = None,
+                      in_port = None, out_port = None, in_ports = None, out_ports = None, priority = None, action = None,
+                      ip_src = None, ip_src_mask = None, ip_dst = None, ip_dst_mask= None, ip_proto = None,
+                      ingress_mirror_id = None, egress_mirror_id = None, counter = None, redirect_port = None):
+
+        stage = SAI_ACL_STAGE_INGRESS
+        priority = SAI_SWITCH_ATTR_ACL_TABLE_MINIMUM_PRIORITY
```

Comment:
there are two priorities, table priority and entry priority. currently you are using priority for both, should be separated.

table_priority = value of SAI_SWITCH_ATTR_ACL_TABLE_MINIMUM_PRIORITY
entry_priority = vaule of SAI_SWITCH_ATTR_ACL_ENTRY_MINIMUM_PRIORITY

---
In reply to: [78723210](https://github.com/opencomputeproject/SAI/pull/148#discussion_r78723210) [](ancestors = 78723210)